src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -658,81 +625,154 @@ private static ISymbol GetEventAccessor(IEventSymbol iEvent, IOperation operatio
+            }
+            else if (symbol is IEventSymbol iEvent)
+            {
+                AnalyzeEventAccessor(operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, iEvent);
+            }
+            else if (symbol is IMethodSymbol method)
+            {
+                AnalyzeTypeParameter(operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, method);
+            }
+
+            static void CheckTypeArguments(ImmutableArray<ITypeSymbol> namedTypes, IOperation operation, OperationAnalysisContext context,
+                PooledConcurrentDictionary<KeyValuePair<IOperation, ISymbol>, SmallDictionary<string, PlatformAttributes>> platformSpecificOperations,
+                ConcurrentDictionary<ISymbol, SmallDictionary<string, PlatformAttributes>?> platformSpecificMembers, ImmutableArray<string> msBuildPlatforms)
+            {
+                foreach (var typeArgument in namedTypes)
+                {
+                    if (typeArgument.SpecialType == SpecialType.None)
+                    {
+                        CheckImmediateAttributes(operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, typeArgument);
+
+                        if (typeArgument is INamedTypeSymbol nType && nType.IsGenericType)
+                        {
+                            foreach (var tArgument in nType.TypeArguments)
+                            {
+                                if (tArgument.SpecialType == SpecialType.None)
+                                {
+                                    CheckImmediateAttributes(operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, tArgument);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            static void CheckImmediateAttributes(IOperation operation, OperationAnalysisContext context, PooledConcurrentDictionary<KeyValuePair<IOperation, ISymbol>,
+                SmallDictionary<string, PlatformAttributes>> platformSpecificOperations, ConcurrentDictionary<ISymbol, SmallDictionary<string, PlatformAttributes>?> platformSpecificMembers,
+                ImmutableArray<string> msBuildPlatforms, ISymbol symbol)
+            {
+                if (TryGetOrCreateImmediatePlatformAttributes(symbol, platformSpecificMembers, out var operationAttributes))
+                {
+                    CheckWithCallSiteAttributes(operation, context, platformSpecificOperations, platformSpecificMembers, msBuildPlatforms, symbol, operationAttributes);
+                }
+            }
+
+            static void AnalyzePropertyAccessors(IOperation operation, OperationAnalysisContext context, PooledConcurrentDictionary<KeyValuePair<IOperation, ISymbol>,
```

Comment:
Feels like you can also revert these new local functions to reduce the noise in this PR - instead just change the original code to pass in `immediateOnly: true` to `CheckOperationAttributes` for accessors.